src/frontend/src/hooks/AuthHooks.ts

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from 'react';
+import { Nullable } from 'types/types';
+import {ApiError } from './types';
+import { supabase } from 'main';
+import { Session } from '@supabase/supabase-js';
+
+export default function useAuth() {
+    const [isLoading, setLoading] = useState<boolean>(true);
+    const [session, setSession] = useState<Session | null>();
+    const [errors, _] = useState<Nullable<ApiError>>(null);
+
+    useEffect(() => {
+        setLoading(true);
+        const fetchSession = async () => {
+            supabase.auth.getSession().then(({ data: { session } }) => {
+                setSession(session);
+            });
+            setLoading(false);
```

Comment:
We should also make not of auth errors here,

```suggestion
        if (response.ok) {
            // I haven't double checked this payload package
            // Here, I would have expected a response.json() call though
            setSession(response.data); 
        } else {
            // we should define a new type accrodingly
            const errorData = data as AuthServerError; 
            const authError = { error: errorData.title, message: errorData.errors, user: userName };
            console.error(JSON.stringify(authError));
            setErrors(authError);
        }
            setLoading(false);
```

Reply:
We don't need to do this, and we don't want the session as the data, session has is own object and this is what is recommended by Supabase.